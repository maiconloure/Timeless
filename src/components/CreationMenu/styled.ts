import styled from 'styled-components';

export const Menu = styled.div`
  background: #8ac0e9;
  width: 260px;
  max-width: 400px;
  color: #fff;
  padding-bottom: 1px;

  &:active {
    cursor: grabbing;
  }
`;

export const MenuSection = styled.div`
  background-color: #fff;
  margin: 10px;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
`;

export const SectionTitle = styled.div`
  font-family: Fira Code;
  font-style: normal;
  font-weight: bold;
  font-size: 1.5rem;
  line-height: 1.6rem;
  background-color: var(--color-primary-4);
  padding: 2px 4px;
  border-radius: 2px 2px 0px 0px;

  h4 {
    margin: 2px;
  }
`;

export const MenuTitle = styled.div`
  background-color: var(--color-primary-1);
  text-align: right;
  span {
    vertical-align: middle;
    display: inline-block;
    margin: 4px;
    height: 20px;
    width: 20px;
    border-radius: 100%;
  }

  span:nth-child(1) {
    cursor: pointer;
    background-color: #f59439;
  }
  span:nth-child(2) {
    cursor: pointer;
    background-color: #0c395c;
  }
  span:nth-child(3) {
    cursor: pointer;
    background-color: #da0000;
  }
`;

export const MenuOptions = styled.div<{ selectedCard: any }>`
  display: flex;
  align-items: center;
  img {
    cursor: pointer;
    width: 55px;
    padding: 4px;
    margin: 0 2px;
    cursor: pointer;
  }

  img#card:active {
    -webkit-filter: drop-shadow(5px 5px 5px gray);
    filter: drop-shadow(5px 5px 5px gray);
    transform: scale(1.2);
  }

  ${(props) =>
    props.selectedCard.removeCard &&
    ` img#trash {
      -webkit-filter: drop-shadow(5px 5px 5px gray);
      filter: drop-shadow(5px 5px 5px gray);
      transform: scale(1.2);
  }`}

  ${(props) =>
    props.selectedCard.fastCard &&
    ` img#fast {
      -webkit-filter: drop-shadow(5px 5px 5px gray);
      filter: drop-shadow(5px 5px 5px gray);
      transform: scale(1.2);
  }`}

${(props) =>
    props.selectedCard.addText &&
    ` img#addText {
      -webkit-filter: drop-shadow(5px 5px 5px gray);
      filter: drop-shadow(5px 5px 5px gray);
      transform: scale(1.2);
  }`}

${(props) =>
    props.selectedCard.connect &&
    ` img#connect {
      -webkit-filter: drop-shadow(5px 5px 5px gray);
      filter: drop-shadow(5px 5px 5px gray);
      transform: scale(1.2);
  }`}

${(props) =>
    props.selectedCard.followedCard &&
    ` img#pin {
      -webkit-filter: drop-shadow(5px 5px 5px gray);
      filter: drop-shadow(5px 5px 5px gray);
      transform: scale(1.2);
  }`}
  

${(props) =>
    props.selectedCard.blockedCard &&
    ` img#blocked {
      -webkit-filter: drop-shadow(5px 5px 5px gray);
      filter: drop-shadow(5px 5px 5px gray);
      transform: scale(1.2);
  }`}
`;
