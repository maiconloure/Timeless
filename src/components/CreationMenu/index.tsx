import React from 'react';
import { icons } from 'utils';

import { CreationMenuProps } from '../ComponentsInterface';
import * as St from './styled';

const CreationMenu = ({
  selectedCard,
  groupButton,
  createCardButton,
  createFasterCardButton,
  removeCardButton,
  createTextButton,
  connectArrowButton,
  desconnectArrowButton,
  pinCardButton,
  blockCardButton,
  className,
  cards,
}: CreationMenuProps) => (
  <St.Menu className={className}>
    <St.MenuTitle>
      <span />
      <span />
      <span />
    </St.MenuTitle>

    <St.MenuSection>
      <St.SectionTitle>
        <h4>OBJETOS</h4>
      </St.SectionTitle>

      <St.MenuOptions selectedCard={selectedCard}>
        {/* <div className="tooltip">
          <img id="group" src={icons.group} onClick={groupButton} alt="group" />
          <span className="tooltiptext">Agrupar cartões</span>
        </div> */}

        <div className="tooltip">
          <img id="card" src={icons.card} onClick={createCardButton} alt="create" />
          <span className="tooltiptext">Criar cartão</span>
        </div>

        <div className="tooltip">
          <img id="fast" src={icons.fastCard} onClick={createFasterCardButton} alt="fast" />
          <span className="tooltiptext">Cartão Rápido</span>
        </div>

        <div className="tooltip">
          <img id="trash" src={icons.trash} onClick={removeCardButton} alt="remove" />
          <span className="tooltiptext">Habilitar lixeira</span>
        </div>
      </St.MenuOptions>
    </St.MenuSection>

    <St.MenuSection>
      <St.SectionTitle>
        <h4>FERRAMENTAS</h4>
      </St.SectionTitle>

      <St.MenuOptions selectedCard={selectedCard}>
        {/* <div className="tooltip">
          <img id="addText" src={icons.addText} onClick={createTextButton} alt="text" />
          <span className="tooltiptext">Adicionar Texto</span>
        </div> */}

        <div className="tooltip">
          <img id="connect" src={icons.connect} onClick={connectArrowButton} alt="connect" />
          <span className="tooltiptext">Habilitar modo conexão</span>
        </div>
        <div className="tooltip">
          <img id="card" src={icons.desconnect} onClick={desconnectArrowButton} alt="connect" />
          <span className="tooltiptext">Desfazer última conexão</span>
        </div>

        <div className="tooltip">
          <img id="pin" src={icons.pin} onClick={pinCardButton} alt="pin/follow" />
          <span className="tooltiptext">Seguir cartão</span>
        </div>

        <div className="tooltip">
          <img id="blocked" src={icons.blocked} onClick={blockCardButton} alt="block" />
          <span className="tooltiptext">Bloquear cartão</span>
        </div>
      </St.MenuOptions>
    </St.MenuSection>
  </St.Menu>
);

export default CreationMenu;
