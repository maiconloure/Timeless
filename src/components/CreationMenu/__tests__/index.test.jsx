import React from 'react';
import { useDispatch } from 'react-redux';
import renderer from 'react-test-renderer';

import CreationMenu from '../index';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(() => jest.fn()),
  useSelector: jest.fn(() => ({})),
}));

jest.mock('../../../utils/importAll', () => ({ icons: {} }));

describe('Snapshot', () => {
  it('renders', () => {
    const tree = renderer.create(<CreationMenu selectedCard={{}} setSelectedCard={{}} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
