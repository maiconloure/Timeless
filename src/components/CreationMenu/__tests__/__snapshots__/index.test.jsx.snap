// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders 1`] = `
<div
  className="sc-bdnylx gOyGLT"
>
  <div
    className="sc-hKFyIo gnBfdK"
  >
    <span />
    <span />
    <span />
  </div>
  <div
    className="sc-gtssRu bgBLvI"
  >
    <div
      className="sc-dlnjPT csVGFK"
    >
      <h4>
        OBJETOS
      </h4>
    </div>
    <div
      className="sc-eCApGN leDWvJ"
    >
      <div
        className="tooltip"
      >
        <img
          alt="group"
          id="group"
        />
        <span
          className="tooltiptext"
        >
          Agrupar cartões
        </span>
      </div>
      <div
        className="tooltip"
      >
        <img
          alt="create"
          id="card"
        />
        <span
          className="tooltiptext"
        >
          Cartão padrão
        </span>
      </div>
      <div
        className="tooltip"
      >
        <img
          alt="fast"
          id="fast"
        />
        <span
          className="tooltiptext"
        >
          Cartão Rápido
        </span>
      </div>
      <div
        className="tooltip"
      >
        <img
          alt="remove"
          id="trash"
        />
        <span
          className="tooltiptext"
        >
          Habilitar lixeira
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-gtssRu bgBLvI"
  >
    <div
      className="sc-dlnjPT csVGFK"
    >
      <h4>
        FERRAMENTAS
      </h4>
    </div>
    <div
      className="sc-eCApGN leDWvJ"
    >
      <div
        className="tooltip"
      >
        <img
          alt="text"
          id="addText"
        />
        <span
          className="tooltiptext"
        >
          Adicionar Texto
        </span>
      </div>
      <div
        className="tooltip"
      >
        <img
          alt="connect"
          id="connect"
        />
        <span
          className="tooltiptext"
        >
          Conectar cartão
        </span>
      </div>
      <div
        className="tooltip"
      >
        <img
          alt="pin/follow"
          id="pin"
        />
        <span
          className="tooltiptext"
        >
          Seguir cartão
        </span>
      </div>
      <div
        className="tooltip"
      >
        <img
          alt="block"
          id="blocked"
        />
        <span
          className="tooltiptext"
        >
          Bloquear cartão
        </span>
      </div>
    </div>
  </div>
</div>
`;
